src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -2327,6 +2333,198 @@
+                        {
+                            // Check to see if this breakpoint has a condition that has changed.
+                            if (!StringComparer.Ordinal.Equals(ad7BPRequest.Condition, b.Condition))
+                            {
+                                // Condition has been modified. Delete breakpoint so it will be recreated with the updated condition.
+                                var toRemove = m_dataBreakpoints[b.DataId];
+                                toRemove.Delete();
+                                m_dataBreakpoints.Remove(b.DataId);
+                            }
+                            else
+                            {
+                                if (ad7BPRequest.BindResult != null)
+                                {
+                                    response.Breakpoints.Add(ad7BPRequest.BindResult);
+                                }
+                                else
+                                {
+                                    response.Breakpoints.Add(new Breakpoint()
+                                    {
+                                        Id = (int)ad7BPRequest.Id,
+                                        Verified = false,
+                                        Line = 0
+                                    });
+
+                                }
+                                continue;
+                            }
+                        }
+                    }
+
+                    // Bind the new data bp
+                    if (!m_dataBreakpoints.ContainsKey(b.DataId))
+                    {
+                        int hr = HRConstants.S_OK;
+
+                        int lastCommaIdx = b.DataId.LastIndexOf(',');
+                        if (lastCommaIdx == -1)
+                        {
+                            eb.ThrowHR(HRConstants.E_FAIL);
+                        }
+
+                        // format is "{dataId},{size}" where dataId = "{address},{displayName}"
+                        string strSize = b.DataId.Substring(lastCommaIdx + 1);
+                        string address = b.DataId.Substring(0, lastCommaIdx);
+                        uint size = uint.Parse(strSize, CultureInfo.InvariantCulture);
+
+                        AD7BreakPointRequest pBPRequest = new AD7BreakPointRequest(address, size);
+                        hr = m_engine.CreatePendingBreakpoint(pBPRequest, out IDebugPendingBreakpoint2 pendingBp);
+                        eb.CheckHR(hr);
+
+                        hr = pendingBp.Bind();
+                        if (hr == HRConstants.S_OK)
+                        {
+                            newBreakpoints[b.DataId] = pendingBp;
+                        }
+                        response.Breakpoints.Add(pBPRequest.BindResult);
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+            }
```

Comment:
>} [](http://example.com/codeflow?start=12&length=1)

I think you want something like `responder.SetError(ex.Message)` and move the `SetResponse` to the end of the try.